base/server/upgrade/10.6.0/02-FixServerConfiguration.py

```diff
@@ -35,10 +35,11 @@ def upgrade_instance(self, instance):
             'catalina.properties',
             '/usr/share/pki/server/conf/catalina.properties')
 
-        self.replace_with_link(
-            instance,
-            'ciphers.info',
-            '/usr/share/pki/server/conf/ciphers.info')
+        if os.path.exists('/usr/share/pki/server/conf/ciphers.info'):
+            self.replace_with_link(
+                instance,
+                'ciphers.info',
+                '/usr/share/pki/server/conf/ciphers.info')
```

Comment:
This script will only be executed if you're upgrading from version `10.6.0` or earlier. If you're upgrading from `10.8.3` you'll need a new upgrade script for the target version (i.e. `10.10.0`).

If people no longer need to configure ciphers in `server.xml` we might not need this file anymore. Should we just remove the link?

Reply:
Yeah, I was going to add that in another upgrade script, probably the same one as above.

Reply:
I suppose we'll have an upgrade script to remove `ciphers.info` in 10.10.0. The change in 10.6.0 upgrade script is probably not necessary, but should be harmless.

Reply:
Yeah but I'd hate for someone to start at Fedora 26 or 28 (unpatched) and try to move to F32 (patched). They'd end up going through 10.6's upgrade scripts... Probably? If they start on 10.5? Not sure :-)

Reply:
This particular one is probably fine, but in general I'd avoid changing an old upgrade script since it's harder to test them and there might be newer scripts that depend specific changes made in that script. It's like going back in time and change an event in the past which causes a ripple effect (probably watched too much sci-fi).

New changes generally should be handled with new upgrade scripts for the current version. At some point in the future we might want to drop older upgrade scripts, e.g. upgrading from anything older than 10.5 will no longer supported.